apps/frontend/src/app/2_molecules/DashboardWelcomeBanner/DashboardWelcomeBanner.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { FC } from 'react';
+
+import { SupportedTokens } from '@sovryn/contracts';
+import { Button } from '@sovryn/ui';
+
+import { useWalletConnect } from '../../../hooks';
+import { useAssetBalance } from '../../../hooks/useAssetBalance';
+
+export type DashboardWelcomeBannerProps = {
+  openLOC: () => void;
+  connectWallet: () => void;
+};
+
+export const DashboardWelcomeBanner: FC<DashboardWelcomeBannerProps> = ({
+  openLOC,
+  connectWallet,
+}) => {
+  const { account } = useWalletConnect();
+  const { value, loading } = useAssetBalance(SupportedTokens.rbtc);
```

Comment:
Maybe split this component to children components - one for non-connected user and one for connected user.
This way you could have `useAssetBalance` inside of child component and prevent rpc call for balance check on non-existing wallet